### ISSUE TITLE
exponential integral (Ei, E1, En...) function

### ISSUE THREAD

**Issue by stevengj** [JuliaLang/julia#7089]
This is a [common special function](http://en.wikipedia.org/wiki/Exponential_integral) that it would be nice to include.  It already is supplied by MPFR, which gives us a `BigFloat` version.

**Comment by stevengj** [comment]
I assigned the problem of implementing `E₁` to double-precision accuracy in the right-half complex plain (`real(z) ≥ 0`) in problem set 3 of [our course 18.S096 this January](https://github.com/stevengj/18S096-iap17).   [In the solutions](https://github.com/stevengj/18S096-iap17/blob/master/pset3/pset3-solutions.ipynb), I showed how to implement it using a combination of Taylor series and continued fractions, with some custom macros to do inlining, and got performance 5–6 times faster than the Fortran code used in SciPy.

This should be helpful as a starting point for more full-featured implementations.
